### PR TITLE
[MODEL-12684] CustomTasks Network Access: docs and examples

### PR DESCRIPTION
**Pull request by eric-s-s**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary

This adds some docs and examples for custom tasks network access functionality

### CODE REVIEW

**Comment by yakov-g** [comment]
jenkins run test_integration_general

**Review by yakov-g — APPROVED** [review]

**Comment by yakov-g** [comment]
jenkins run continuous-integration/jenkins/pr-merge

File: MODEL-METADATA.md [hunk 1]
```diff
@@ -38,7 +38,18 @@ which label will be chosen as the predicted label.
 ## Options specific to tasks
 * trainOnProject (optional): A hash with the pid of a project you would like to train your new model or version 
 on. If this is supplied, the code you supplied will start to run against this pid automagically. 
+* userCredentialSpecifications (optional): This is a list of credentials that will be injected from
```
Comment (scottp-ml) [inline MODEL-METADATA.md:41]: You will also want to make a docs ticket to have this added to the custom task documentation.

File: task_templates/2_estimators/13_python_credentials_and_internet_access/model-metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      condition: IN
+      value:
+        - NUM
+    - field: number_of_columns
+      condition: NOT_LESS_THAN
+      value: 2
+    - field: sparse
+      condition: EQUALS
+      value: SUPPORTED
+# Since this is a fake example, ignoring this requirement so that we can use test data
```
Comment (scottp-ml) [inline task_templates/2_estimators/13_python_credentials_and_internet_access/model-metadata.yaml:31]: maybe just remove this instead of commenting out

**Review by scottp-ml — APPROVED** [review]

**Pull request merged**